src/linux_maps.rs

```diff
@@ -51,49 +51,96 @@ pub fn get_process_maps(pid: Pid) -> std::io::Result<Vec<MapRange>> {
     // Parses /proc/PID/maps into a Vec<MapRange>
     let maps_file = format!("/proc/{}/maps", pid);
     let mut file = File::open(maps_file)?;
+
+    // Check that the file is not too big
+    let metadata = file.metadata()?;
+    if metadata.len() > 0x1000000 {
```

Comment:
Can you clarify what "too big" means in this case? Is this based on an actual limit in the kernel?

Reply:
It's defensive and arbitrary. If feel risky, I can enlarge it limitation or remove it.

Reply:
The default configuration for Linux is 65530 vmas, each vma make take up to 4k+ bytes at most. So seems we need to enlarge the constraint to 0x10000000 at least.

Reply:
updated